samples/storage/src/main/java/com/example/platform/storage/mediastore/SelectedPhotosAccess.kt

```diff
@@ -0,0 +1,266 @@
+    val context = LocalContext.current
+    val lifecycleOwner = LocalLifecycleOwner.current
+    val coroutineScope = rememberCoroutineScope()
+    var files by remember { mutableStateOf(emptyList<FileEntry>()) }
+
+    val storageAccess by produceState(
+        initialValue = StorageAccess.Denied,
+        context,
+        lifecycleOwner,
+    ) {
+        val eventObserver = LifecycleEventObserver { _, event ->
+            if (event == Lifecycle.Event.ON_RESUME) {
+                value = getStorageAccess(context)
+                coroutineScope.launch {
+                    files = getVisualMedia(context.contentResolver)
+                }
+            }
+        }
+
+        lifecycleOwner.lifecycle.addObserver(eventObserver)
+        awaitDispose {
+            lifecycleOwner.lifecycle.removeObserver(eventObserver)
+        }
+    }
+
+    val requestPermissions = rememberLauncherForActivityResult(RequestMultiplePermissions()) {
+        // Better logic to handle denied & permanently denied access should be written here.
+        // We recommend you to look at the {Single Permission} sample
+    }
+
+    Column(Modifier.fillMaxSize()) {
+        ListItem(
+            headlineContent = { Text("Storage Access") },
+            trailingContent = { Text(storageAccess.name) },
+        )
+        Divider()
+        ListItem(
+            headlineContent = { Text("Add files to the selection") },
+            trailingContent = {
+                if (files.isNotEmpty()) {
+                    Text("${files.size} items")
+                }
+            },
+            supportingContent = {
+                if (storageAccess == StorageAccess.Full) {
+                    Text("Access to library fully granted")
+                } else {
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
+                        Row(horizontalArrangement = Arrangement.spacedBy(5.dp)) {
+                            TextButton(
+                                onClick = { requestPermissions.launch(arrayOf(READ_MEDIA_IMAGES)) },
+                            ) {
+                                Text("Images")
+                            }
+                            TextButton(
+                                onClick = { requestPermissions.launch(arrayOf(READ_MEDIA_VIDEO)) },
+                            ) {
+                                Text("Videos")
+                            }
+                            TextButton(
+                                onClick = {
+                                    requestPermissions.launch(
+                                        arrayOf(READ_MEDIA_IMAGES, READ_MEDIA_VIDEO),
+                                    )
+                                },
+                            ) {
+                                Text("Both")
+                            }
+                        }
+                    } else if (Build.VERSION.SDK_INT == Build.VERSION_CODES.TIRAMISU) {
```

Comment:
I read this like in T you should request both permissions and in U you should split them.

I think you should show the split from T onwards but in U add the READ_MEDIA_VISUAL_USER_SELECTED in the request.

Reply:
Merging U & T logic makes sense but requesting `READ_MEDIA_VISUAL_USER_SELECTED` isn't useful here